src/main/java/com/google/devtools/build/lib/rules/cpp/CcLibrary.java

```diff
@@ -333,16 +334,16 @@ public static void init(
           }
         }
 
-        // If no DEF file is specified and the windows_export_all_symbols feature is enabled, parse
-        // object files to generate DEF file and use it to export symbols - if we have a parser.
-        // Otherwise, use no DEF file.
-        if (defFile == null
-            && CppHelper.shouldUseGeneratedDefFile(ruleContext, featureConfiguration)) {
-          defFile = generatedDefFile;
-        }
-
-        if (defFile != null) {
-          linkingHelper.setDefFile(defFile);
+        // 1. If a custom DEF file is specified in win_def_file attribute, use it.
```

Comment:
This decision logic is the same as in cc_library, can you extract it to a function?

Reply:
Done,